File: backend/src/main/java/com/quizmatrix/quizmatrix/model/Quiz.java
package com.quizmatrix.quizmatrix.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity(name = "Quiz")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Quiz {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private Integer id_quiz;
    private Integer id_domain;
    private String title;
    private String description;
    private java.sql.Date creation_date;
    private Integer time;
}

File: frontend/QuizMatrix/src/app/models/quiz.ts
import { Question } from './question';

export class Quiz {
  id_quiz: number;
  id_domain: number;
  title: string;
  description: string;
  creation_date: Date;
  time: number;
  questions: Question[] = [];

  constructor(
    id_quiz: number,
    id_domain: number,
    title: string,
    description: string,
    creation_date: Date,
    time: number
    // questions: Question[]
  ) {
    this.id_quiz = id_quiz;
    this.id_domain = id_domain;
    this.title = title;
    this.description = description;
    this.creation_date = creation_date;
    this.time = time;
    // this.questions = questions;
  }
}
